src/controllers/mainMenuController.js

```diff
@@ -42,44 +42,83 @@ var DetailedMenuController = function ($, wot, localization, gmApi, settings) {
             '#site-report': goToSiteReport,
             '#is-filter': onIsFilterChange
         };
-        Object.keys(menuEvents).forEach(function (item) {
+        Object.keys(menuEvents).forEach(function(item) {
             $(contentDocument.querySelectorAll(item)).on('click', menuEvents[item]);
         });
     };
 
-    var onIsFilterChange = function () {
+    var onIsFilterChange = function() {
         var isFilter = contentDocument.getElementById('is-filter').checked;
+
+        // animate class for prevent animation while the state from the application is determined
+        $(contentDocument.querySelectorAll(".menu-filter_label")).addClass("animate");
+        
         showHideBlockAdButton(isFilter);
+        setFilteringStateToStore(isFilter);
         gmApi.ADG_changeFilteringState(isFilter);
     };
 
-    var setInitFilteringState = function () {
-        gmApi.ADG_isFiltered(function (isFiltered) {
-            var input = contentDocument.getElementById('is-filter');
+    var setInitFilteringState = function() {
+        var input = contentDocument.getElementById('is-filter');
+        input.checked = getFilteringStateFromStore();
+        gmApi.ADG_isFiltered(function(isFiltered) {
             input.checked = isFiltered;
+            setFilteringStateToStore(isFiltered);
+            showHideBlockAdButton(isFiltered);
```

Comment:
Why don't you call `showHideBlockAdButton` immediately after you've read filtering state from the local storage?